### PR TITLE
Proposed fix for #237

### PR DESCRIPTION
**Pull request by Synchro**
Resubmission of a clean version of #306

### CODE REVIEW

**Comment by Synchro** [comment]
The build failure is an unrelated composer problem.

**Comment by oliverklee** [comment]
I'll have a look at the composer problem.

**Comment by oliverklee** [comment]
Hi @Synchro, could you also please add regression tests for this (i.e., unit tests that fail without the fix and that pass with it)? Thanks!

**Comment by Synchro** [comment]
This test works for me - but another started failing. It turned out to be because it contained invalid markup so I fixed that.

**Comment by oliverklee** [comment]
@Synchro I've just fixed the Composer-related breakage on master. Could you please rebase, force-push and then comment here? Thanks!

**Comment by Synchro** [comment]
I've rebased, squashed and pushed.

File: Classes/Emogrifier.php [hunk 1]
```diff
@@ -924,21 +924,26 @@ private function copyCssWithMediaToStyleNode(\DOMDocument $xmlDocument, \DOMXPat
...
-        preg_match_all('#(?<query>@media[^{]*\\{(?<css>(.*?)\\})(\\s*)\\})#s', $css, $mediaQueries);
+        preg_match_all('/@media\b[^{]*({((?:[^{}]+|(?1))*)})/', $css, $mediaQueries, PREG_SET_ORDER);
         $result = [];
-        foreach (array_keys($mediaQueries['css']) as $key) {
-            $result[] = [
-                'css' => $mediaQueries['css'][$key],
-                'query' => $mediaQueries['query'][$key],
-            ];
+        foreach ($mediaQueries as $query) {
+            if (!empty(trim($query[2]))) {
```
Comment (oliverklee) [inline Classes/Emogrifier.php]: You'll need to put trim(...) into a local variable so the empty call also works in PHP 5.4:
https://travis-ci.org/jjriv/emogrifier/jobs/128679215#L181

**Comment by Synchro** [comment]
Should work now!

**Comment by oliverklee** [comment]
Hi @Synchro, I've polished your PR a bit and pushed it as #319. Could you please check the polished version and comment there? Thanks!

**Pull request closed without merging**